plugins/inputs/vsphere/vsan.go

```diff
@@ -0,0 +1,539 @@
+						bucket = metricEntry{name: mn, ts: ts, fields: make(map[string]interface{}), tags: tags}
+						buckets[bKey] = bucket
+					}
+					if v, err := strconv.ParseFloat(values, 32); err == nil {
+						bucket.fields[metricLabel] = v
+					}
+				}
+			}
+			if len(timeStamps) > 0 {
+				lastSample := timeStamps[len(timeStamps)-1]
+				if lastSample != (time.Time{}) && lastSample.After(latest) {
+					latest = lastSample
+				}
+			}
+			// We've iterated through all the metrics and collected buckets for each measurement name. Now emit them!
+			for _, bucket := range buckets {
+				acc.AddFields(bucket.name, bucket.fields, bucket.tags, bucket.ts)
+			}
+			count += len(buckets)
+		}
+	}
+	e.hwMarks.Put(hwMarksKeyPrefix+clusterRef.ref.Value, latest)
+	return nil
+}
+
+// queryDiskUsage adds 'FreeCapacityB' and 'TotalCapacityB' metrics to telegraf accumulator
+func (e *Endpoint) queryDiskUsage(ctx context.Context, vsanClient *soap.Client, clusterRef objectRef, acc telegraf.Accumulator) error {
+	resp, err := vsanmethods.VsanQuerySpaceUsage(ctx, vsanClient,
+		&vsantypes.VsanQuerySpaceUsage{
+			This:    spaceManagerRef,
+			Cluster: clusterRef.ref,
+		})
+	if err != nil {
+		return err
+	}
+	fields := make(map[string]interface{})
+	fields["free_capacity_byte"] = resp.Returnval.FreeCapacityB
+	fields["total_capacity_byte"] = resp.Returnval.TotalCapacityB
+	tags := populateClusterTags(make(map[string]string), clusterRef, e.URL.Host)
+	acc.AddFields(vsanSummaryMetricsName, fields, tags)
+	return nil
+}
+
+// queryDiskUsage adds 'OverallHealth' metric to telegraf accumulator
+func (e *Endpoint) queryHealthSummary(ctx context.Context, vsanClient *soap.Client, clusterRef objectRef, acc telegraf.Accumulator) error {
+	fetchFromCache := true
+	resp, err := vsanmethods.VsanQueryVcClusterHealthSummary(ctx, vsanClient,
+		&vsantypes.VsanQueryVcClusterHealthSummary{
+			This:           healthSystemRef,
+			Cluster:        clusterRef.ref,
+			Fields:         []string{"overallHealth", "overallHealthDescription"},
+			FetchFromCache: &fetchFromCache,
+		})
+	if err != nil {
+		return err
+	}
+	healthStr := resp.Returnval.OverallHealth
+	healthMap := map[string]int{"red": 2, "yellow": 1, "green": 0}
+	fields := make(map[string]interface{})
+	if val, ok := healthMap[healthStr]; ok {
+		fields["overall_health"] = val
+	}
+	tags := populateClusterTags(make(map[string]string), clusterRef, e.URL.Host)
+	acc.AddFields(vsanSummaryMetricsName, fields, tags)
+	return nil
+}
+
+// queryResyncSummary adds resync information to accumulator
+func (e *Endpoint) queryResyncSummary(ctx context.Context, vsanClient *soap.Client, clusterObj *object.ClusterComputeResource, clusterRef objectRef, acc telegraf.Accumulator) error {
+	if lower, _ := versionLowerThan(e.apiVersion, "6.7"); lower {
+		log.Printf("I! [inputs.vsphere][vSAN] Minimum API Version 6.7 required for resync summary. Found: %s. Skipping VCenter: %s", e.apiVersion, e.URL.Host)
+		return nil
+	}
```

Comment:
First of all please check the potential error. Second, please just remove the call to this function instead of logging this over and over. Third, do not use `log`.